### PR TITLE
adding ip-restriction feat

### PR DESCRIPTION
**Pull request by tdaroly**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following appy to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/app/resource-manager/Microsoft.App/stable/2022-05-01/ContainerApps.json [hunk 1]
```diff
@@ -756,6 +766,41 @@
...
+      "description": "Rules to restrict incoming IP address.",
+      "type": "object",
+      "required": [
+        "name",
+        "ipAddress",
+        "action"
+      ],
+      "properties": {
+        "name": {
+          "description": "Name for the ip restriction rule.",
```
Comment (vinisoto) [inline specification/app/resource-manager/Microsoft.App/stable/2022-05-01/ContainerApps.json:779]: nit: IP (same comment in other places)

**Review by vinisoto — APPROVED** [review]
Minor casing comments

**Review by vinisoto — CHANGES_REQUESTED** [review]
Changes should go to /preview/2022-06-01-preview, not to /stable/2022-05-01

File: specification/app/resource-manager/Microsoft.App/stable/2022-05-01/ContainerApps.json (hunk 1 above)
Comment (tdaroly) [reply specification/app/resource-manager/Microsoft.App/stable/2022-05-01/ContainerApps.json]: done

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ContainerApps.json [hunk 2]
```diff
@@ -756,6 +766,41 @@
...
+        "description": {
+          "description": "Describe the IP restriction rules that are being sent to the container-app. This is an optional field.",
+          "type": "string"
+        },
+        "ipAddress": {
+          "description": "CIDR notation to match incoming IP address",
+          "type": "string"
+        },
+        "action": {
+          "description": "Allow or Deny rules to determine for incoming IP",
```
Comment (jijohn14) [inline specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ContainerApps.json:791]: We can add a note that only one action type is allowed for complete rule set

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/examples/ContainerApps_CreateOrUpdate.json [hunk 3]
```diff
@@ -30,6 +30,20 @@
                 "revisionName": "testcontainerApp0-ab1234",
                 "label": "production"
               }
+            ],
+            "ipSecurityRestrictionRules": [
+              {
+                "name": "name",
```
Comment (jijohn14) [inline specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/examples/ContainerApps_CreateOrUpdate.json:36]: Better name and desc in example :)

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/examples/ContainerApps_CreateOrUpdate.json [hunk 4]
```diff
@@ -30,6 +30,20 @@
...
+                "description": "desc",
+                "ipAddress": "192.168.1.1/32",
+                "action": "Allow"
+              },
+              {
+                "name": "name",
+                "description": "desc",
+                "ipAddress": "192.168.1.1/16",
+                "action": "Deny"
```
Comment (jijohn14) [inline specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/examples/ContainerApps_CreateOrUpdate.json:45]: I believe deny is not possible in the same rule set

Comment (tdaroly) [reply specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/examples/ContainerApps_CreateOrUpdate.json]: fixed

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/examples/ContainerApps_CreateOrUpdate.json (hunk 3 above)
Comment (tdaroly) [reply specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/examples/ContainerApps_CreateOrUpdate.json]: updated

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ContainerApps.json (hunk 2 above)
Comment (tdaroly) [reply specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ContainerApps.json]: added

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ContainerApps.json [hunk 5]
```diff
@@ -641,6 +641,16 @@
         "allowInsecure": {
           "description": "Bool indicating if HTTP connections to is allowed. If set to false HTTP connections are automatically redirected to HTTPS connections",
           "type": "boolean"
+        },
+        "ipSecurityRestrictionRules": {
```
Comment (ruslany) [inline specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ContainerApps.json:645]: >ipSecurityRestrictionRules [](http://example.com/codeflow?start=9&length=26)

Is this name consistent with other RPs? Please check other RPs such as app service, SQL, etc to see what names they use.

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ContainerApps.json (hunk 2 above)
Comment (ruslany) [reply specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ContainerApps.json]: Please confirm that this schema is consistent with what other Azure RPs use.

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/examples/ContainerApps_CreateOrUpdate.json (hunk 3 above)
Comment (ruslany) [reply specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/examples/ContainerApps_CreateOrUpdate.json]: This needs to be included in the response examples too.

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ContainerApps.json (hunk 5 above)
Comment (tdaroly) [reply specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ContainerApps.json]: Renamed to maintain consistency with other services
MicrosoftTeams-image

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ContainerApps.json [hunk 6]
```diff
@@ -756,6 +766,41 @@
         }
       }
     },
+    "IpSecurityRestrictionRule": {
+      "description": "Rules to restrict incoming IP address.",
```
Comment (ruslany) [inline specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ContainerApps.json:770]: >Rules [](http://example.com/codeflow?start=22&length=5)

Rule

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ContainerApps.json [hunk 7]
```diff
@@ -756,6 +766,41 @@
...
+        "ipAddress",
+        "action"
+      ],
+      "properties": {
+        "name": {
+          "description": "Name for the IP restriction rule.",
+          "type": "string"
+        },
+        "description": {
+          "description": "Describe the IP restriction rules that are being sent to the container-app. This is an optional field.",
```
Comment (ruslany) [inline specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ContainerApps.json:783]: >rules [](http://example.com/codeflow?start=54&length=5)

rule

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ContainerApps.json [hunk 8]
```diff
@@ -756,6 +766,41 @@
...
+          "type": "string"
+        },
+        "ipAddress": {
```
Comment (ruslany) [inline specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ContainerApps.json:786]: >ipAddress [](http://example.com/codeflow?start=9&length=9)

Can we change the name to ipAddressRange? I know it is not consistent with appService but that is more correct name (appservice just could not change the name after API was released). If this is a too big change then keeping as it is ok.

**Comment by rkmanda** [comment]
>     "title": "ContainerApps API Client"

pl address the breaking changes

---
Refers to: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ContainerApps.json:5 in 3025a6f. [](commit_id = 3025a6ff0b5348f5573af7d9ccc9ca770cddd122, deletion_comment = False)

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ContainerApps.json (hunk 8 above)
Comment (tdaroly) [reply specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ContainerApps.json]: renamed

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ContainerApps.json (hunk 6 above)
Comment (ruslany) [reply specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ContainerApps.json:770]: The description is still not fixed

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ContainerApps.json (hunk 7 above)
Comment (ruslany) [reply specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ContainerApps.json:783]: This is still not fixed

**Review by ruslany — APPROVED** [review]
:shipit:

**Comment by tdaroly** [comment]
@jorgecotillo Can you please merge the PR ? I do not have write access to the repo.

**Review by lirenhe — APPROVED** [review]

**Pull request merged**